src/components/date-picker/date-picker.js

```diff
@@ -0,0 +1,204 @@
+   */
+  constructor(element, options) {
+    super(element, options);
+    const type = this.element.getAttribute(this.options.attribType);
+    if (type === 'no-calendar') {
+      this._addInputLogic(this.element.querySelector(this.options.selectorDatePickerInput));
+    } else {
+      this.element.calendar = this._initDatePicker(type);
+    }
+  }
+
+  _initDatePicker = (type) => {
+    const date = (type === 'single')
+    ? this.element.querySelector(this.options.selectorDatePickerInput)
+    : this.element.querySelector(this.options.selectorDatePickerInputFrom);
+    const calendar = new Flatpickr(date, Object.assign(flattenOptions(this.options), {
+      allowInput: true,
+      mode: type,
+      onClose: (selectedDates) => {
+        this._updateInputFields(selectedDates, type);
+      },
+      onChange: () => {
+        this._updateClassNames(calendar);
+      },
+      onMonthChange: () => {
+        this._updateClassNames(calendar);
+      },
+      onYearChange: () => {
+        this._updateClassNames(calendar);
+      },
+      onOpen: () => {
+        this._updateClassNames(calendar);
+      },
+      nextArrow: this._rightArrowHTML(),
+      prevArrow: this._leftArrowHTML(),
+    }));
+    if (type === 'range') {
+      this.element.querySelector(this.options.selectorDatePickerInputTo).addEventListener('click', () => {
+        this.element.querySelector(this.options.selectorDatePickerInputTo).focus();
+        calendar.open();
+      });
+      this._addInputLogic(this.element.querySelector(this.options.selectorDatePickerInputTo));
+    }
+    this.element.querySelector(this.options.selectorDatePickerIcon).addEventListener('click', () => {
+      calendar.open();
+    });
+    this._addInputLogic(date);
+    return calendar;
+  }
+
+  getCalendar = () => this.element.calendar;
+
+  setMinDate = (date) => {
+    this.element.calendar.set('minDate', date);
+  }
+
+  setMaxDate = (date) => {
+    this.element.calendar.set('maxDate', date);
+  }
+
+  setDisabledDates = (dates) => {
+    this.element.calendar.set('disable', dates);
+  }
+
+  setDefaultDates = (dates) => {
+    this.element.calendar.set('defaultDate', dates);
+  }
+
+  _rightArrowHTML() {
+    return (`
+      <svg width="8" height="12" viewBox="0 0 8 12" fill-rule="evenodd">
+        <path d="M0 10.6L4.7 6 0 1.4 1.4 0l6.1 6-6.1 6z"></path>
+      </svg>`
+    );
+  }
+
+  _leftArrowHTML() {
+    return (`
+      <svg width="8" height="12" viewBox="0 0 8 12" fill-rule="evenodd">
+        <path d="M7.5 10.6L2.8 6l4.7-4.6L6.1 0 0 6l6.1 6z"></path>
+      </svg>`
+    );
+  }
+
+  _addInputLogic = (input) => {
+    const inputField = input;
+    inputField.addEventListener('focus', () => {
+      inputField.value = inputField.value.replace(/\s+/g, '');
+    });
+    inputField.addEventListener('change', () => {
+      if (inputField.value.length === 10) {
+        const inputDate = new Date(inputField.value);
```

Comment:
Letting `Date` constructor to parse date will cause globalization issues. I'd suggest using `parseDate(string, format)` API of Flatpickr to align to the parse logic Flatpickr runs internally.